loki/transformations/inline.py

```diff
@@ -415,7 +415,7 @@ def _map_unbound_dims(var, val):
         check for check in FindInlineCalls().visit(callee.body) if check.function == 'PRESENT'
     )
     present_map = {
-        check: sym.Literal('.true.') if check.arguments[0] in call.arg_map else sym.Literal('.false.')
+        check: sym.Literal('.true.') if check.arguments[0] in [arg.name for arg in call.arg_map] else sym.Literal('.false.')
```

Comment:
I think this might now fail if the casing of the variable spelling doesn't match between present check and argument. You might want to use something like this:
```suggestion
        check: sym.Literal('.true.')
        if str(check.arguments[0]).lower() in {arg.name.lower() for arg in call.arg_map}
        else sym.Literal('.false.')
```
Pylint also flagged the line length here

Reply:
I added an specific test in test_inline.py

This is quite funny, I just tried and leaving it as is, the presence check evaluation is completely case-insensitive and this works whatever the casing used in the routine.

Reply:
Ah, yes that makes sense!